python/tank/util/filesystem.py

```diff
@@ -343,3 +343,64 @@ def create_valid_filename(value):
         # so that we return a string
         u_src = value.decode("utf-8")
         return exp.sub("_", u_src).encode("utf-8")
+
+def get_permissions(path):
+    """
+    Retrieve the file system permissions for the file or folder in the
+    given path.
+
+    :param filename: Path to the file to be queried for permissions
+    :returns: permissions bits of the file 
+    :raises: OSError - if there was a problem retrieving permissions for the path
+    """
+    return stat.S_IMODE(os.lstat(path)[stat.ST_MODE])
+
+def delete_folder(path):
+    """
+    Deletes a folder and all of its contents recursively, even if it has read only items
+
+    :param path: File system path to location to the folder to be deleted
+    :returns: True if the folder and all of its contents were deleted without
```

Comment:
I think the bit that says 
>  Problems deleting any items will be reported as errors in the log output but otherwise ignored and skipped; meaning the function will continue deleting as much as it can
should be a sphinx `.. note::`